int16 count
int16 turns
bool ccw
